ui/src/app/views/run/test/table/test.table.component.ts

```diff
@@ -35,34 +35,33 @@ export class TestTableComponent extends Table {
 
     updateFilteredTests(): void {
         this.filteredTests = new Array<TestCase>();
+        if (!this.tests) {
+            return;
+        }
         switch (this.filter) {
             case 'error':
-                if (this.tests) {
-                    this.tests.forEach(ts => {
-                        if (ts.errors > 0 || ts.failures > 0) {
-                            this.filteredTests.push(...ts.tests.map(tc => {tc.name = ts.name + ' / ' + tc.name; return tc}).filter(tc => {
-                                return (tc.errors && tc.errors.length > 0) || (tc.failures && tc.failures.length > 0);
-                            }));
-                        }
-                    });
-                }
+                for (let ts of this.tests) {
+                    if (ts.errors > 0 || ts.failures > 0) {
+                        let testCases = ts.tests
+                        .filter(tc => {return (tc.errors && tc.errors.length > 0) || (tc.failures && tc.failures.length > 0)})
```

Comment:
no need to write `{ return ... }` because when it's inlined the return is implicit so it's the same than 
`.filter(tc => (tc.errors && tc.errors.length > 0) || (tc.failures && tc.failures.length > 0))`